src/components/Referrals/referralsHelper.js

```diff
@@ -66,6 +66,21 @@ export const tierDiscountInfo = {
   2: 10,
 };
 
+export function deserializeSampleStats(input) {
+  const parsedData = JSON.parse(input);
```

Comment:
could you check please what will happen in case there is bad value store in local storage? e.g. incorrect json. will useLocalStorage fail in this case?

we should treat localstorage as a user input in general and handle incorrect input so the app still works

Reply:
and also handle case when there is empty object or some fields are missing. 
so the goal is to not fail in case of any data there

Reply:
Sure!

Reply:
mmm i guess in case input is incorrect json this code will fail? i think we should at least wrap it in try/catch

Reply:
is there a chance that in case some fields are missing the code that uses it will throw error?

Reply:
We are adding the data in local storage in an Array.  You can look at the updated code below. This will check the data to be an array and the different values inside array to be an object. 

```js
export function deserializeSampleStats(input) {
  const parsedData = JSON.parse(input);
  if (!Array.isArray(parsedData)) return [];
  return parsedData.map((data) => {
    if (typeof data !== "object") return {};
    return Object.keys(data).reduce((acc, cv) => {
      const currentValue = data[cv];
      if (currentValue?.type === "BigNumber") {
        acc[cv] = bigNumberify(currentValue.hex || 0);
      } else {
        acc[cv] = currentValue;
      }
      return acc;
    }, {});
  });
}
```

Reply:
i mean if i add manually incorrect value to localstorage e.g. `[{time: Date.now()}]` then page will be broken
<img width="993" alt="image" src="https://user-images.githubusercontent.com/95585424/173231880-0c18bda5-d28c-4158-8cab-63aaf827eb01.png">

localstorage data should be verified as any other user input like query params so it can't break the page. it's just some thing to keep in mind when we're using some value that user can change directly and validate it by default.
tbh i'm not 100% sure if every other handles that, but hope it does